### PR TITLE
OT-CreateReportListView | Created a Reports table to show the users data in the reports page

### LINKED ISSUE #21: OT-CreateReportListView
Create a list view that appears when you scroll down on the reports page. On the figma you will need to implement the reports under PROOF OF CONCEPT. Ignore making the navbar that says "hello user" and the sub navbar that says "Sunday -> Monday -> Tuesday"

Create the front-end based off of our mockup on figma: https://www.figma.com/file/kSZKeE1vutPwQwUM6oDvwL/Off-Top?node-id=0%3A1

### PR DESCRIPTION
**Pull request by steve545**
**What I did:**

- I created a scrollable  table in the reports page to show user reports.

**How to test:**

- Start the zookeeper server.

- Start the kafka server.

- Start Java spring-boot.

- Make sure port forwarding is active.

- Start up the OFF-TOP app.

- Go to the Reports page.

- The Reports table should be shown along with the user reports and the table should be vertically scrollable.

Capture-1

### CODE REVIEW

File: lib/user_report_table.dart [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+              const Text('Focus',
+                  style: TextStyle(fontSize: 19, color: Colors.black)),
+            ]),
+          ),
+        ]));
+        columnTitles = true;
+      }
+
+      rows.add(
+        TableRow(children: [
```
Comment (frlzjosh) [inline lib/user_report_table.dart]: These containers can be moved into a stateless function to get rid of duplicate code. In this function make it of type `Widget` and you can pass two parameters, one being the color, and one is the data
```
Widget getTableData(data, dataColor){
   Container(
            padding: const EdgeInsets.fromLTRB(8.0, 6.0, 8.0, 6.0),
            child: Column(children: [
              Text(color,
                  style: const TextStyle(fontSize: 17, color: dataColor)),
            ]),
          ),
}
```
and then now you can do 
```
rows.add(
   getTableData(rowData.date, Colors.black),
   getTableData(rowData.duration, Colors.black),
   getTableData(rowData.focus, getFocusColor(rowData.focus))
)
```

Comment (frlzjosh) [reply lib/user_report_table.dart]: Move getTableData into  components/reports/getTableDataWidget.dart

File: lib/user_report_table.dart [hunk 2]
```diff
@@ -0,0 +1,142 @@
+import 'package:flutter/cupertino.dart';
```
Comment (frlzjosh) [inline lib/user_report_table.dart]: move the file into components/reports/userReportTable.dart

**Review by frlzjosh — CHANGES_REQUESTED** [review]
looking good! There are some things that need to moved around and refactored.

File: lib/components/footer/bottomNavigationTabs.dart [hunk 3]
```diff
@@ -3,6 +3,8 @@ import 'package:off_top_mobile/login.dart';
 import 'package:off_top_mobile/recordingSession.dart';
 import 'package:off_top_mobile/settings_page.dart';
 
+import '../../reports.dart';
```
Comment (tonynguyen98) [inline lib/components/footer/bottomNavigationTabs.dart]: Can you change this import of this to `package:off_top_mobile/reports.dart` in the event where we need to move the file, it'll be easier to change?

File: lib/user_report_table.dart [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            padding: const EdgeInsets.fromLTRB(8.0, 6.0, 8.0, 6.0),
+            child: Column(children: [
+              Text(rowData.focus,
+                  style: TextStyle(
+                      fontSize: 17, color: getFocusColor(rowData.focus))),
+            ]),
+          ),
+        ]),
+      );
+    }
```
Comment (tonynguyen98) [inline lib/user_report_table.dart]: Can you move all these out of the build Widget and into it's own Widget?

**Review by tonynguyen98 — DISMISSED** [review]
Just some changes, the app works currently

**Review by frlzjosh — APPROVED** [review]
looks good! Great job!

**Pull request merged**